src/components/SignUp/GeneralInfo.js

```diff
@@ -1,29 +1,437 @@
+        return (
+          <View style={styles.form} key={idx}>
+            {Title}
+            {Input}
+          </View>
+        );
+      }
+    );
+
+    return Forms;
+  }
+
+  // If a form has several inputs, the title should be able to add new input set
+  getTitle(title) {
+    if (title.isArray == false) {
+      return <Text style={styles.title}>{title.name}</Text>;
+    }
+
+    return (
+      <View style={styles.flexR}>
+        <View style={styles.horiL}>
+          <Text style={styles.title}>{title.name}</Text>
         </View>
+        <TouchableWithoutFeedback onPress={() => this.addNewItem(title.name)}>
+          <View style={styles.horiR}>
+            <Text style={styles.add}>{'+ Add item'}</Text>
+          </View>
+        </TouchableWithoutFeedback>
       </View>
     );
   }
 
+  // Add new empty input set.
+  addNewItem(listName) {
+    if (listName == 'Education') {
+      this.state.education.unshift({
+        school: { name: '' },
+        type: '',
+        year: { name: '' },
+        concentration: [ { name: '' } ],
+      });
+
+      this.setState({
+        eduDataSource: this.state.eduDataSource.cloneWithRows(this.state.education),
+      });
+    } else if (listName == 'Experience') {
+      this.state.experience.unshift({
+        start_date: '',
+        end_date: '',
+        employer: { name: '' },
+        location: { name: '' },
+        position: [ { name: '' } ],
+      });
+
+      this.setState({
+        workDataSource: this.state.workDataSource.cloneWithRows(this.state.experience),
+      });
+    }
+  }
+
+  getInput(title) {
+    let Input = null;
+    let propName = title.name.toLowerCase();
+    if (title.isArray) {
+      let list = this.state[propName];
+      if (title.name == 'Education') {
+        let renderRow = (edu, sectionID, rowID) =>
+                          this.getDefaultEdu(edu, sectionID, rowID);
+        Input = (
+          <ListView
+            key={this.state.education}
+            enableEmptySections={true}
+            dataSource={this.state.eduDataSource}
+            renderRow={renderRow}/>
+        );
+      } else if (title.name == 'Experience') {
+        let renderRow = (edu, sectionID, rowID) =>
+                          this.getDefaultWork(edu, sectionID, rowID);
+        Input = (
+          <ListView
+            key={this.state.experience}
+            enableEmptySections={true}
+            dataSource={this.state.workDataSource}
+            renderRow={renderRow}/>
+        );
+      }
+    } else {
+      let defaultValue = this.state[propName];
+      Input = this.getTextInput(propName, defaultValue);
+    }
+
+    return Input;
+  }
+
+  getDefaultEdu(edu, sectionID, rowID) {
+    let eduName = edu.school === undefined ? '' : edu.school.name;
+    let startYear = edu.startYear === undefined ? '' : edu.startYear.name;
+    let endYear = edu.year === undefined ? '' : edu.year.name;
+    if (edu.endYear !== undefined) endYear = edu.endYear.name;
```

Comment:
When I sign in first, there is no start and end fields. So, I think checking is nesseary.